src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Bjorhus.cpp

```diff
@@ -0,0 +1,488 @@
+    const auto negative_lambda0 =
+        get(dot_product(normal_covector, *face_mesh_velocity));
+    for (size_t a = 0; a < 4; ++a) {
+      char_speeds.at(a) -= negative_lambda0;
+    }
+  }
+
+  Bjorhus::add_constraint_preserving_terms_to_dt_v_psi(
+      make_not_null(&bc_dt_v_psi), unit_interface_normal_vector,
+      three_index_constraint, char_speeds);
+
+  Bjorhus::add_constraint_preserving_terms_to_dt_v_zero(
+      make_not_null(&bc_dt_v_zero), unit_interface_normal_vector,
+      four_index_constraint, char_speeds);
+
+  if (type_ == detail::ConstraintPreservingBjorhusType::ConstraintPreserving) {
+    Bjorhus::add_constraint_preserving_terms_to_dt_v_minus(
+        make_not_null(&bc_dt_v_minus), gamma2, coords, incoming_null_one_form,
+        outgoing_null_one_form, incoming_null_vector, outgoing_null_vector,
+        projection_ab, projection_Ab, projection_AB,
+        char_projected_rhs_dt_v_psi, char_projected_rhs_dt_v_minus,
+        constraint_char_zero_plus, constraint_char_zero_minus, char_speeds);
+  } else if (type_ == detail::ConstraintPreservingBjorhusType::
+                          ConstraintPreservingPhysical) {
+    Bjorhus::add_constraint_preserving_physical_terms_to_dt_v_minus(
+        make_not_null(&bc_dt_v_minus), gamma2, coords, normal_covector,
+        unit_interface_normal_vector, spacetime_unit_normal_vector,
+        incoming_null_one_form, outgoing_null_one_form, incoming_null_vector,
+        outgoing_null_vector, projection_ab, projection_Ab, projection_AB,
+        inverse_spatial_metric, extrinsic_curvature, spacetime_metric,
+        inverse_spacetime_metric, three_index_constraint,
+        char_projected_rhs_dt_v_psi, char_projected_rhs_dt_v_minus,
+        constraint_char_zero_plus, constraint_char_zero_minus, phi, d_phi, d_pi,
+        char_speeds);
+  } else {
+    ERROR("Failed to set dtVMinus. Input option \""
+          << "\" Must be one of ConstraintPreserving or "
+             "ConstraintPreservingPhysical");
+  }
+
+  // Only add corrections at grid points where the char speeds are negative
+  helpers::set_bc_corr_zero_when_char_speed_is_positive(
+      make_not_null(&bc_dt_v_psi), char_speeds.at(0));
+  helpers::set_bc_corr_zero_when_char_speed_is_positive(
+      make_not_null(&bc_dt_v_zero), char_speeds.at(1));
+  helpers::set_bc_corr_zero_when_char_speed_is_positive(
+      make_not_null(&bc_dt_v_minus), char_speeds.at(3));
+
+  // Convert corrections to dt<evolved variables>
+  auto dt_evolved_vars = evolved_fields_from_characteristic_fields(
+      gamma2, bc_dt_v_psi, bc_dt_v_zero, bc_dt_v_plus, bc_dt_v_minus,
+      normal_covector);
+
+  *dt_pi_correction = get<Tags::Pi<Dim, Frame::Inertial>>(dt_evolved_vars);
+  *dt_phi_correction = get<Tags::Phi<Dim, Frame::Inertial>>(dt_evolved_vars);
+  *dt_spacetime_metric_correction =
+      get<gr::Tags::SpacetimeMetric<Dim, Frame::Inertial, DataVector>>(
+          dt_evolved_vars);
+
+  // Subtract out original dt<vars>
```

Comment:
can this code block be removed or should it be uncommented?

Reply:
removed already